timely_beliefs/beliefs/classes.py

```diff
@@ -303,23 +343,9 @@ def query(
 
         # Apply source filter
         if source is not None:
-            source_list = [source] if not isinstance(source, list) else source
-            id_list = [s for s in source_list if isinstance(s, int)]
-            name_list = [s for s in source_list if isinstance(s, str)]
-            if len(id_list) + len(name_list) < len(source_list):
-                unidentifiable_list = [
-                    s
-                    for s in source_list
-                    if not isinstance(s, int) and not isinstance(s, str)
-                ]
-                raise ValueError(
-                    "Query by source failed: query only possible by integer id or string name. Failed sources: %s"
-                    % unidentifiable_list
-                )
-            else:
-                q = q.join(DBBeliefSource).filter(
-                    (cls.source_id.in_(id_list)) | (DBBeliefSource.name.in_(name_list))
-                )
+            sources: list = [source] if not isinstance(source, list) else source
+            source_cls = sources[0].__class__
```

Comment:
The `source` parameter to this function is of type `Optional[Union[int, List[int], str, List[str]]]`, so I don't think the `__class__` you get here will be anything else than `int` or `str`.

Reply:
Thanks, I just forgot to update the type annotation and docstring. The `__class__` should give the BeliefSource, or whatever subclass is used (like DataSource in FlexMeasures).